syft/internal/fileresolver/unindexed_directory.go

```diff
@@ -33,15 +40,15 @@ type UnindexedDirectory struct {
 	fs   afero.Fs
 }
 
-func NewFromUnindexedDirectory(dir string) file.WritableResolver {
+func NewFromUnindexedDirectory(dir string) DeletingWritableResolver {
```

Comment:
I think all of these constructors should return the concrete type `UnindexedDirectory` and let the caller determine if they want writable/deletable behavior by their variable type.

Reply:
I removed changes to this package because of using `afero`